core/src/commonMain/kotlin/com/xebia/functional/xef/llm/assistants/Assistant.kt

```diff
@@ -62,5 +69,96 @@ class Assistant(
       val response = assistantsApi.createAssistant(request)
       return Assistant(response.body(), assistantsApi, api)
     }
+
+    suspend fun fromConfig(
+      request: String,
+      assistantsApi: AssistantsApi = fromEnvironment(::AssistantsApi),
+      api: AssistantApi = fromEnvironment(::AssistantApi)
+    ): Assistant {
+      val parsed = Yaml.Default.decodeYamlMapFromString(request)
+      val assistantRequest =
+        AssistantRequest(
+          assistantId = parsed["assistant_id"]?.literalContentOrNull,
+          model = parsed["model"]?.literalContentOrNull ?: error("model is required"),
+          name = parsed["name"]?.literalContentOrNull,
+          description = parsed["description"]?.literalContentOrNull,
+          instructions = parsed["instructions"]?.literalContentOrNull,
+          tools =
+            parsed["tools"]?.let { list ->
+              (list as List<*>).map { element ->
+                when (element) {
+                  is Map<*, *> -> {
+                    val tool =
+                      element["type".toYamlElement()]?.toString() ?: error("type is required")
+                    when (tool) {
+                      "code_interpreter" -> AssistantTool.CodeInterpreter
+                      "retrieval" -> AssistantTool.Retrieval
+                      "function" -> {
+                        val name =
+                          element["name".toYamlElement()]?.toString() ?: error("name is required")
+                        val description =
+                          element["description".toYamlElement()]?.toString()
+                            ?: error("description is required")
+                        val parameters =
+                          element["parameters".toYamlElement()]?.toString()
+                            ?: error("parameters is required")
+                        AssistantTool.Function(name, description, parameters)
+                      }
+                      else -> error("unknown tool $tool")
+                    }
+                  }
+                  else -> error("unknown tool $element")
+                }
+              }
+            },
+          fileIds =
+            parsed["file_ids"]?.let { (it as List<*>).map { it.toString() } } ?: emptyList(),
+        )
+      return if (assistantRequest.assistantId != null) {
+        val assistant = Assistant(assistantRequest.assistantId, assistantsApi, api)
+        assistant.modify(
+          ModifyAssistantRequest(
+            name = assistantRequest.name,
+            description = assistantRequest.description,
+            instructions = assistantRequest.instructions,
+            tools = assistantTools(assistantRequest),
+            fileIds = assistantRequest.fileIds,
+            metadata = null // assistantRequest.metadata
```

Comment:
Should we manage the metadata or can we wait for addressing in another PR?

Reply:
Let's address in another PR we don't use it yet and not sure what the best format is